qt/python/mantidqt/widgets/plotconfigdialog/axestabwidget/presenter.py

```diff
@@ -40,10 +42,13 @@ def __init__(self, fig, view=None, parent=None):
             self.update_view)
         self.view.axis_tab_bar.currentChanged.connect(
             self.axis_changed)
-        self.view.apply_all_button.clicked.connect(self.apply_all_properties)
         self.view.show_minor_ticks_check_box.toggled.connect(
             self.show_minor_ticks_checked)
         self.view.autoscale.toggled.connect(self.autoscale_changed)
+        if self.success_callback and self.error_callback:
+            self.view.apply_all_button.clicked.connect(self._apply_all_properties_with_errors)
+        else:
+            self.view.apply_all_button.clicked.connect(self.apply_all_properties)
```

Comment:
This matches the established convention in the file, but we shouldn't have [signal / slot connections in the view](https://developer.mantidproject.org/Standards/GUIStandards.html#signals-setup-in-the-view-to-the-presenter-only)

Reply:
Ah I wasn't aware of that. So these `.connect` calls should actually be happening in the view, and what we have now is essentially the presenter going in and connecting stuff on behalf of the view?

Reply:
Yeah spot on really, the view should either take a lambda or an observer (preferred) to call back in the presenter
That way you know that the lambda or observer will call the method directly, and you don't have to worry about testing each side of the Qt signal/slot in the presenter